src/coreclr/nativeaot/System.Private.CoreLib/src/System/InvokeUtils.cs

```diff
@@ -128,121 +128,334 @@ private static Exception ConvertOrWidenPrimitivesEnumsAndPointersIfPossible(obje
             }
 
             EETypeElementType dstElementType = dstEEType->ElementType;
-            if (!CanPrimitiveWiden(dstElementType, srcEEType->ElementType))
-            {
-                dstObject = null;
-                return CreateChangeTypeArgumentException(srcEEType, dstEEType);
-            }
-
+            EETypeElementType srcElementType = srcEEType->ElementType;
+
+            void* rawDstValue = null;
+
+            Unsafe.SkipInit(out char charValue);
+            Unsafe.SkipInit(out sbyte sbyteValue);
+            Unsafe.SkipInit(out byte byteValue);
+            Unsafe.SkipInit(out short shortValue);
+            Unsafe.SkipInit(out ushort ushortValue);
+            Unsafe.SkipInit(out int intValue);
+            Unsafe.SkipInit(out uint uintValue);
+            Unsafe.SkipInit(out long longValue);
+            Unsafe.SkipInit(out ulong ulongValue);
+            Unsafe.SkipInit(out float floatValue);
+            Unsafe.SkipInit(out double doubleValue);
+
+            Unsafe.SkipInit(out dstObject);
+
+            ref byte rawSrcValue = ref RuntimeHelpers.GetRawData(srcObject);
+
+            // This is the table of all supported widening conversions:
+            //
+            // Boolean (W = BOOL)
+            // Char (W = U2, CHAR, I4, U4, I8, U8, R4, R8)
+            // SByte (W = I1, I2, I4, I8, R4, R8)
+            // Byte (W = CHAR, U1, I2, U2, I4, U4, I8, U8, R4, R8)
+            // Int16 (W = I2, I4, I8, R4, R8)
+            // UInt16 (W = U2, CHAR, I4, U4, I8, U8, R4, R8)
+            // Int32 (W = I4, I8, R4, R8)
+            // UInt32 (W = U4, I8, U8, R4, R8)
+            // Int64 (W = I8, R4, R8)
+            // UInt64 (W = U8, R4, R8)
+            // Single (W = R4, R8)
+            // Double (W = R8)
```

Comment:
```suggestion
```
I do not think this comment is useful. It is easy to see what is supported by reading the code below, and the matrix is not particularly tricky.

Reply:
Wasn't sure whether it'd be ok to remove that too along with the method. Done in ea0289d4a5c834b241fd31959cec97295bc450b2.